UT4MasterServer/Controllers/UnrealTournamentMatchmakingController.cs

```diff
@@ -235,8 +235,29 @@ public async Task<IActionResult> UpdateGameServerPlayers(string id, [FromBody] G
 		if (server.OwningClientID != user.Session.ClientID)
 			Unauthorized();
 
-		server.PublicPlayers = serverOnlyWithPlayers.PublicPlayers;
-		server.PrivatePlayers = serverOnlyWithPlayers.PrivatePlayers;
+		// handle player list update
+		foreach (var player in serverOnlyWithPlayers.PublicPlayers)
+		{
+			if (!server.PublicPlayers.Where(x => x == player).Any())
+			{
+				server.PublicPlayers.Add(player);
+			}
+			if (server.PrivatePlayers.Where(x => x == player).Any())

```

Comment:
else

Reply:
The way i wrote it is the intended logic. I want to add it to `PublicPlayers` if it doesn't exist there yet and i want to remove it from `PrivatePlayers` if it exists there. This way we ensure that there is only one instance of some player across both arrays.

Reply:
On the first sight, it looked to me that they were both same list.